ExtractUserDataActivity/notification.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * An extensions of io-functions-commons/dist/src/models/notification to implement missing query methods
+ * Ideally they will be integrated in the common module
+ */
+
+import * as DocumentDb from "documentdb";
+import {
+  NotificationModel as NotificationModelCommons,
+  RetrievedNotification
+} from "io-functions-commons/dist/src/models/notification";
+import * as DocumentDbUtils from "io-functions-commons/dist/src/utils/documentdb";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+
+export class NotificationModel extends NotificationModelCommons {
+  /**
+   * Creates a new Notification model
+   *
+   * @param dbClient the DocumentDB client
+   * @param collectionUrl the collection URL
+   */
+  constructor(
+    dbClient: DocumentDb.DocumentClient,
+    collectionUrl: DocumentDbUtils.IDocumentDbCollectionUri
+  ) {
+    super(dbClient, collectionUrl);
+  }
+
+  /**
+   * Returns the messages for the provided fiscal code
+   *
+   * @param fiscalCode The fiscal code of the recipient
+   */
+  public findNotificationsForRecipient(
+    fiscalCode: FiscalCode
+  ): DocumentDbUtils.IResultIterator<RetrievedNotification> {
+    return DocumentDbUtils.queryDocuments(
+      this.dbClient,
+      this.collectionUri,
+      {
+        parameters: [
+          {
+            name: "@fiscalCode",
+            value: fiscalCode
+          }
+        ],
+        query: `SELECT * FROM m WHERE m.fiscal_code = @fiscalCode`
```

Comment:
since the partitionKey of the notifications collection is `messageId` this results in a cross partition query which should be avoided (see https://docs.microsoft.com/it-it/azure/cosmos-db/how-to-query-container#cross-partition-query). 

you should include a contidion on `messageId` so: get all users messages then, for each retrieved message, loop to get the message-status and the relative nofitications; then, for each retrieved notification, loop to get then the notification-status.